.github/workflows/build_and_test.yml

```
@@ -297,6 +298,79 @@ jobs:
           fail_ci_if_error: false
           verbose: true
 
+  macos-m1:
+    runs-on: ${{ matrix.os-version }}
+    strategy:
+      matrix:
+        os-version: [macos-14]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v5
+        with:
+          # support version: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
+          python-version: "3.12"
+
+      - name: Configure PcapPlusPlus
+        # Ensure user have access to network devices instead of giving super-user right
+        # Seems macos-14 has more strict file permission
+        run: |
+          sudo chmod a+rw /dev/bpf*
+          sudo chmod a+rw /usr/local
+          cmake -DPCAPPP_BUILD_COVERAGE=ON -S . -B "$BUILD_DIR"
+
+      - name: Build PcapPlusPlus
+        run: cmake --build "$BUILD_DIR" -j
+
+      - name: Check architecture
+        run: lipo $BUILD_DIR/Pcap++/libPcap++.a -verify_arch ${{ matrix.arch }}
```

you need to remove this check or hard code the arch

```suggestion
        run: lipo $BUILD_DIR/Pcap++/libPcap++.a -verify_arch arm64
       ```

---

ops, forgot this.